src/cpu/pred/btb/microtage.cc

```
@@ -0,0 +1,1079 @@
+
+        tableIndexBits[i] = ceilLog2(tableSizes[i]);
+        tableIndexMasks[i].resize(tableIndexBits[i], true);
+
+        assert(histLengths.size() >= numPredictors);
+
+        assert(tableTagBits.size() >= numPredictors);
+        tableTagMasks[i].resize(tableTagBits[i], true);
+
+        assert(tablePcShifts.size() >= numPredictors);
+
+        tagFoldedHist.push_back(PathFoldedHist((int)histLengths[i], (int)tableTagBits[i], 16));
```

Cppcheck (2.19.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)